gemrb/GUIScripts/GUIClasses.py

```diff
@@ -54,152 +66,292 @@ class GSymbol:
     'Unload': _GemRB.Symbol_Unload
   }
 
-class GWindow:
-  __metaclass__ = metaIDWrapper
+class Scrollable:
+	def Scroll(self, x, y, relative = True):
+		_GemRB.Scrollable_Scroll(self, x, y, relative)
+
+	def ScrollTo(self, x, y):
+		self.Scroll(x, y, False)
+
+	def ScrollDelta(self, x, y):
+		self.Scroll(x, y, True)
+
+class GView:
+	__metaclass__ = metaIDWrapper
+	methods = {
+	'AddAlias': _GemRB.View_AddAlias,
+	'AddSubview': _GemRB.View_AddSubview,
+	'SetEventProxy': _GemRB.View_SetEventProxy,
+	'GetFrame': _GemRB.View_GetFrame,
+    'SetFrame': _GemRB.View_SetFrame,
+    'SetBackground': _GemRB.View_SetBackground,
+    'SetFlags': _GemRB.View_SetFlags,
+	'SetResizeFlags': _GemRB.View_SetResizeFlags,
+	'SetTooltip': _GemRB.View_SetTooltip,
+	'Focus': _GemRB.View_Focus
+	}
+
+	__slots__ = ['SCRIPT_GROUP', 'Flags']
+	
+	def __eq__(self, rhs):
+		if rhs == None:
+			return self.ID == -1
+		return self.ID == rhs.ID and self.SCRIPT_GROUP == rhs.SCRIPT_GROUP
+	
+	def __hash__(self):
+		return self.SCRIPT_GROUP + str(self.ID)
+	
+	def __nonzero__(self):
+		return self.ID != -1
+
+	def GetSize(self):
+		frame = self.GetFrame()
+		return (frame['w'], frame['h'])
+
+	def GetPos(self):
+		frame = self.GetFrame()
+		return (frame['x'], frame['y'])
+
+	def SetSize(self, w, h):
+		r = self.GetFrame()
+		r['w'] = w
+		r['h'] = h
+		self.SetFrame(r);
+
+	def SetPos(self, x, y):
+		r = self.GetFrame()
+		r['x'] = x
+		r['y'] = y
+		self.SetFrame(r);
+
+	def GetInsetFrame(self, l = 0, r = None, t = None, b = None):
+		r = r if r is not None else l
+		t = t if t is not None else l
+		b = b if b is not None else t
+        
+		frame = self.GetFrame()
+		frame['x'] = l
+		frame['y'] = b
+		frame['w'] -= (l + r)
+		frame['h'] -= (b + t)
+		return frame
+
+	def SetVisible(self, visible):
+		self.SetFlags(IE_GUI_VIEW_INVISIBLE, OP_NAND if visible else OP_OR)
+		
+	def IsVisible(self):
+		return (self.Flags&IE_GUI_VIEW_INVISIBLE) != 0
+
+	def SetDisabled(self, disable):
+		self.SetFlags(IE_GUI_VIEW_DISABLED, OP_OR if disable else OP_NAND)
+
+	def CreateControl(self, id, type, x, y, w, h, *args): # backwards compatibility
+		frame = {"x" : x, "y" : y, "w" : w, "h" : h}
+		return self.CreateSubview(id, type, frame, *args)
+
+	def ReplaceSubview(self, subview, ctype, *args):
+		if type(subview) == int:
+			subview = self.GetControl (subview)
+		id = subview.ID & 0x00000000ffffffff
+		frame = subview.GetFrame()
+		RemoveView(subview, True)
+
+		return self.CreateSubview(id, ctype, frame, args)
+
+	def CreateSubview(self, id, type, frame, *args):
+		view = CreateView(id, type, frame, *args) # this will create an entry in the generic 'control' group
+		created = self.AddSubview(view) # this will move the reference into the our window's group
+		RemoveScriptingRef(view) # destroy the old reference just in case something tries to recycle the id while 'created' is still valid
+		return created
+
+	def RemoveSubview(self, view, delete=False):
```

Comment:
Codacy found an issue: [Method could be a function](https://app.codacy.com/manual/gemrb/gemrb/pullRequest?prid=6310305)